homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -36,7 +36,7 @@
 GATEWAY_PLATFORMS = ["alarm_control_panel", "light", "sensor", "switch"]
 SWITCH_PLATFORMS = ["switch"]
 FAN_PLATFORMS = ["fan"]
-HUMIDIFIER_PLATFORMS = ["humidifier", "sensor"]
+HUMIDIFIER_PLATFORMS = ["humidifier", "sensor", "select", "switch", "number"]
```

Comment:
Please only update one platform per PR.

Reply:
> 
> 
> Please only update one platform per PR.

Sorry, I will stick to this rule for future PR's.

Reply:
It will take longer time to review this PR than if we would split it into three PRs, one per platform, and review them.

Reply:
I will split up the PR for switch and number platform. And scope PR for sensor improvements (existing platform) and select.

Reply:
Updates for switch platform: https://github.com/home-assistant/core/pull/52172
Updates for number platform: https://github.com/home-assistant/core/pull/52173